### DISCUSSION TITLE
Implicit constant arguments

### DISCUSSION THREAD

**Discussion by chtenb** [koka-lang/koka#585] [Q&A] [answered]
Suppose we have the following function

```koka
fun monoid/join(v : vector<m>, ?empty : m, ?append : (m, m) -> m) : m
  with pretend-no-div
  var result := @implicit/empty
  v.foreach fn(m)
    result := append(result, m)
  result
```

How can we call this function with implicit parameters?

```koka
pub fun int/add/empty()
  0
pub fun int/add/append(a : int, b : int)
  a + b
  
pub fun main()
  join([1,2,3].vector)
```

This doesn't work, because `int/add/empty` is declared as a function.
As far as I could find, it is currently not possible to declare constant values.
Is there a way to call `monoid/join` without passing the arguments explicitly?

Of course it is possible to use `empty : () -> m` as parameter, but I'm not sure if passing a function has performance downsides compared to just passing a constant value. Besides, it is possible to declare implicit constant parameters, so I would expect it also be possible to use them.

**Comment by TimWhiting** [comment, marked as answer]
You should be able to use `pub val int/add/empty = 0`.

Also see the implicit examples in the tests here: https://github.com/koka-lang/koka/tree/dev/test/overload

**Reply by chtenb** [reply]
Hah, I thought I tried that, but you're right, it works.
